### PR TITLE
Rename "NetSdkManaged" types to "FileBased"

### LINKED ISSUE #23758: Rename "NetSdkManaged" types to "FileBased"
For example, `NetSdkManagedInstaller` should be renamed to `FileBasedInstaller`.

We updated the terminology we use but never renamed the types.  Renaming the types should make the code base easier to understand and read.

### PR DESCRIPTION
**Pull request by gkulin**
Fixes #23758

### CODE REVIEW

**Review by dsplaisted — APPROVED** [review]

**Review by joeloff — APPROVED** [review]

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=23955) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=23955) before we can accept your contribution.<br/><br/>:x: gkulin [sign now](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=23955)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/sdk?pullRequest=23955) it.</sub>

**Comment by dsplaisted** [comment]
@dotnet/nuget-team We had a test fail with the following error:

> /private/tmp/helix/working/B17F0976/p/d/sdk/6.0.300-ci/NuGet.targets(564,5): error : The process cannot access the file '/private/tmp/helix/working/B17F0976/w/A8740934/e/testExecutionDirectory/.dotnet/.nuget/NuGet/nugetorgadd.trk' because it is being used by another process. [/private/tmp/helix/working/B17F0976/w/A8740934/e/testExecutionDirectory/It_publishes_---B5C78A02/Library/Library.csproj]

Do you know what the nugetoradd.trk file is?  Is there possibly a bug in NuGet with concurrent access to that file?

**Comment by joeloff** [comment]
> @dotnet/nuget-team We had a test fail with the following error:
> 
> > /private/tmp/helix/working/B17F0976/p/d/sdk/6.0.300-ci/NuGet.targets(564,5): error : The process cannot access the file '/private/tmp/helix/working/B17F0976/w/A8740934/e/testExecutionDirectory/.dotnet/.nuget/NuGet/nugetorgadd.trk' because it is being used by another process. [/private/tmp/helix/working/B17F0976/w/A8740934/e/testExecutionDirectory/It_publishes_---B5C78A02/Library/Library.csproj]
> 
> Do you know what the nugetoradd.trk file is? Is there possibly a bug in NuGet with concurrent access to that file?

MSBuild had a similar/related issue: https://github.com/dotnet/msbuild/issues/3925#issuecomment-438745162

**Comment by zivkan** [comment]
`nugetoradd.trk` is to mitigate Chocolately and Windows PowerShell's PowerShellGallery module creating a user-profile `nuget.config` without nuget.org as a package source, which confused many customers using these two tools before using dotnet or Visual Studio for the first time on a clean devbox. For more information: https://github.com/NuGet/Home/issues/11387


Perhaps there's an issue with the implementation.  By any chance, on the machines where it's failing, are they running tests in parallel, without ever having run `dotnet build` or `dotnet restore` once before running the tests?  My guess is that the machine doesn't yet have the user-profile `nuget.config` or `nugetorgadd.trk`, and then running multiple processes at the same time that use NuGet.

**Comment by marcpopMSFT** [comment]
Should we consider adding a dotnet build call to the RunTestsOnHelix.* scripts that runs before our helix tests get launched? That could potentially work around this if @zivkan's assumption is accurate.

**Comment by zivkan** [comment]
`dotnet nuget list source` is sufficient for NuGet to create the default config (and nuget.org tracking file).

File: build/RunTestsOnHelix.sh [hunk 1]
```diff
@@ -8,3 +8,8 @@ export TestExecutionDirectory=$(pwd)/testExecutionDirectory
 mkdir $TestExecutionDirectory
 export DOTNET_CLI_HOME=$TestExecutionDirectory/.dotnet
 cp -a $HELIX_CORRELATION_PAYLOAD/t/TestExecutionDirectoryFiles/. $TestExecutionDirectory/
+
+# call dotnet new so the first run message doesn't interfere with the first test
+dotnet new
+# avoid potetial cocurrency issues when nuget is creating nuget.config
```
Comment (joeloff) [inline build/RunTestsOnHelix.sh:14]: small typo, should this be "concurrency"

Comment (gkulin) [reply build/RunTestsOnHelix.sh]: good catch, I just fixed it

**Pull request merged**